docs/concepts-and-explanations/javatemplate.md

```diff
@@ -17,11 +17,12 @@ More advanced refactoring recipes often require the construction of complex [Los
 ```java
 public class ChangeMethodInvocation extends JavaIsoVisitor<ExecutionContext> {
     private final JavaTemplate template =
-        JavaTemplate.builder("withString(#{any(java.lang.String)}).length()")   // Code Snippet
+        JavaTemplate.builder("withString(#{any(java.lang.String)}).length()")  // Code Snippet
             .javaParser(
-                JavaParser.fromJavaVersion()                                    // Parser
-                    .classpath("example-utils"))                                // Classpath lookup
-            .staticImports("org.example.StringUtils.withString")                // Additional import
+                JavaParser.fromJavaVersion()                                   // Parser
+                    .classpath("example-utils"))                               // Classpath lookup
+            .staticImports("org.example.StringUtils.withString")               // Additional import
+            .doBeforeParseTemplate((String template) -> {})                    // Optional side-effect
```

Comment:
Would it be good to outline both `doAfterVariableSubstitution` and `doBeforeParseTemplate` here? With an example that just prints them, something like `System.out::println`

Reply:
Yes: https://github.com/openrewrite/rewrite-docs/pull/399/commits/2d70bd5fe184846ca7eef8b5c8ba4cfe3007ea76!